File: Backend/PrestationServices/src/main/java/tn/esprit/Entities/Connexion.java
package tn.esprit.Entities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
    public class Connexion {
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Long id;
        @ManyToOne
        private Entreprise entrepriseA;
        @ManyToOne
        private Entreprise entrepriseB;
        @Enumerated(EnumType.STRING)
        private Etat etat;
}

File: Backend/Frontend/src/app/models/Connexion/connexion.ts
import { Entreprise } from "../Entreprise/entreprise";

export class Connexion {
    id!:number;
    entrepriseA!:Entreprise;
    entrepriseB!:Entreprise;
    etat!:Etat;
}
export enum Etat {
    EN_ATTENTE = 'ENATTENTE',
    ACCEPTEE = 'ACCEPTEE',
    REFUSEE = 'REFUSEE',
  }